SelfDriveEnv.py

```diff
@@ -1,240 +1,119 @@
-        self.NEW_TILE_REWARD = cfg.reward['new_tile_reward']
-        self.SAME_TILE_REWARD = cfg.reward['same_tile_reward']
-        self.CRASH_REWARD = cfg.reward['crash_reward']
-        self.traveled = []
-    
-    def set_pos(self, coord):
-        self.pos = coord
-
-    def reset(self):
-        self.pos = cfg.car['position']
-        self.angle = cfg.car['angle']
-        self.center_sensors()
-
-    def set_track(self, track):
-        self.track = track
-    
-    def get_car_tip(self):
-        x, y = self.pos
-        return self.rotate((x+25, y), self.angle, (x+25, y+25))
-
-    def get_car_center(self):
-        x, y = self.pos
-        return x+25, y+25
-
-    def dist(self, x0, y0, x1, y1):
-        return ((x0-x1)**2 + (y0-y1)**2)**.5
-
-    def center_sensors(self):
-        x, y = self.pos
-        center = self.get_car_tip()
-        for sensor in self.sensors:
-            sensor[0], sensor[1] = center
 
-    def update_sensors(self):
-        self.center_sensors()
-        offset = 90
-        for sensor in self.sensors:
-            while True:
-                if self.track.colliding_with(sensor[0], sensor[1]):
-                    break                
-                sensor[0] += 1*cos(-radians(self.angle + sensor[2]))
-                sensor[1] += 1*sin(-radians(self.angle + sensor[2]))
-            x0, y0 = self.get_car_tip()
-            x1, y1, = sensor[0], sensor[1]
-            sensor[3] = self.dist(x0, y0, x1, y1)
-        return np.array([sensor[3] for sensor in self.sensors])
-    
-    def render(self, screen):
-        color = (255, 255, 0)
-        self.blitRotateCenter(screen, self.image, self.pos, self.angle)
-        for sensor in self.sensors:
-            pygame.draw.circle(screen, (255, 0, 0), (sensor[0], sensor[1]), 5)
-    
-    def rotate(self, coord, angle, center):
-        angle = -radians(angle+90)
-        x0, y0 = center
-        x1, y1 = coord    
-        d = self.dist(x0, y0, x1, y1)
-        return (d*cos(angle) + x0, d*sin(angle) + y0)
-    
-    def blitRotateCenter(self, screen, image, topleft, angle):
-        rotated_image = pygame.transform.rotate(image, angle)
-        new_rect = rotated_image.get_rect(center = image.get_rect(
-            topleft = topleft).center)
-        screen.blit(rotated_image, new_rect.topleft)
+class Track(gym.Env): 
+    """OpenAI gym environment simulating a car on a racetrack
 
-    def rotate_image_rect(self, image, angle, pos):
-        x, y, = pos
-        rotated_image = pygame.transform.rotate(image, angle)
-        new_rect = rotated_image.get_rect(center = image.get_rect(
-            center = (x, y)).center)
-        return new_rect
+    Most of the general configuration can be tweaked in config.py. Cars must be
+    added to this track via add_car before training/running. Only one car can
+    currently be added to the track but there are plans for multi agent support.
+    In order to create a new track, call reset(new=True). Tracks are saved
```

Comment:
You could just call reset(True) because the "new" variable is set to whatever the parameter is, but if it's empty, it's defaulted to False.

Reply:
Yep, but I'm using the verbose version here for clarity and just in case we expose other features via arguments.